qiskit_experiments/library/characterization/tphi.py

```diff
@@ -0,0 +1,114 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Tphi Experiment class.
+"""
+
+from typing import List, Optional, Union
+import numpy as np
+
+from qiskit.providers import Backend
+from qiskit.test.mock import FakeBackend
+from qiskit_experiments.framework import Options
+from qiskit_experiments.framework.composite.batch_experiment import BatchExperiment
+from qiskit_experiments.library.characterization import T1, T2Ramsey
+from qiskit_experiments.library.characterization.analysis.tphi_analysis import TphiAnalysis
+
+
+class Tphi(BatchExperiment):
+    r"""Tphi Experiment Class
+
+    # section: overview
+
+        :math:`\Gamma_\varphi` is defined as the rate of pure dephasing
+        or depolarization in the :math:`x - y` plane.
+        We compute :math:`\Gamma_\varphi` by computing :math:`\Gamma_2*`, the transverse relaxation rate,
+        and subtracting :math:`\Gamma_1`, the longitudinal relaxation rate. The pure dephasing time
+        is defined by :math:`T_\varphi = 1/\Gamma_\varphi`. Or more precisely,
+
+        :math:`1/T_\varphi = 1/T_{2*} - 1/2T_1`.
+
+        For more details, see :py:class:`T1` and :py:class:`T2Ramsey`
+
+    # section: analysis_ref
+        :py:class:`TphiAnalysis`
+
+    # section: reference
+        .. ref_arxiv:: 1 1904.06560v5
+
+    # section: tutorial
+        :doc:`/tutorials/tphi_characterization`
+    """
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            delays_t1 (Iterable[float]): Delay times of the t1 experiment in seconds.
+            delays_t2 (Iterable[float]): Delay times of the t2ramsey experiment in seconds.
+        """
+        options = super()._default_experiment_options()
+        options.delays_t1 = None
+        options.delays_t2 = None
+        return options
+
+    def __init__(
+        self,
+        qubit: int,
+        delays_t1: List[Union[List[float], np.array]],
+        delays_t2: List[Union[List[float], np.array]],
+        osc_freq: float = 0.0,
+        backend: Optional[Backend] = None,
+    ):
+        """Initialize the experiment object.
+
+        Args:
+            qubit: the qubit under test
+            delays_t1: delay times of the T1 experiment
+            delays_t2: delay times of the T2* experiment
+            osc_freq: the oscillation frequency induced using by the user for T2Ramsey
+            backend: Optional, the backend on which to run the experiment
+        """
+        # Set experiment options
+        options =  self._default_experiment_options()
```

Comment:
Is there a reason not to do it in a way that's different from the other experiments?

Reply:
If instead I do `self.experiment_options.delays_t1 = delays_t1` , I get the error `AttributeError: 'Tphi' object has no attribute '_experiment_options'`.
Same error if I do `self.set_experiment_options(delays_t1=delays_t1, delays_t2=delays_t2)`.
I believe the reason is that I do the `super.init()` later, so the base class members are not recognized yet.

Reply:
Ok, so you can do it later also here. I'm just not sure if what you're doing is equivalent, so I don't want to play with it.